### DISCUSSION TITLE
GraphQL Kickstart & Spring Security

### DISCUSSION THREAD

**Discussion by danielshiplett** [graphql-java-kickstart/graphql-spring-boot#806] [Q&A]
I have an application that has been using Kickstart Spring Boot & Spring Security successfully with query and mutation resolvers for quite a while.  We are on Kickstart 12.0.0 and the latest Spring Boot (2.6.6).  We use Spring's OAuth2 support for our authentication and authorization.  

Just the other day, we started using subscriptions.  Our WebSockets are functioning just fine.  However, we aren't getting the user's security context in the subscription handler.  The context's authentication object is null.

We didn't do anything specifically for the queries or mutations to get it hooked into Spring Security (that I'm aware of).  It just worked.  I can't find any documented information on Kickstart about wiring in Spring Security or why the WebSocket servlet would be different.

Any pointers in getting WebSocket's to have the user's security context would be great!

Thanks!

**Comment by ermadmi78** [comment]
You can see [here](https://github.com/ermadmi78/kobby-gradle-example) an example of secure WebSocket based on Kickstart Spring Boot & Spring Security.
